crates/router/src/connector/fiserv/transformers.rs

```diff
@@ -234,6 +339,48 @@ impl<F, T>
     }
 }
 
+impl<F, T>
+    TryFrom<
+        types::ResponseRouterData<F, FiservPaymentsSyncResponse, T, types::PaymentsResponseData>,
+    > for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<
+            F,
+            FiservPaymentsSyncResponse,
+            T,
+            types::PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        let gateway_resp = match item.response.sync_responses.first() {
+            Some(gateway_response) => gateway_response,
+            _ => Err(errors::ConnectorError::ResponseHandlingFailed)?,
+        };
+
+        Ok(Self {
+            status: gateway_resp
+                .gateway_response
+                .transaction_state
+                .clone()
+                .into(),
```

Comment:
don't use into, call the appropriate the `from` implementation for it. this is help in debugging